app/src/main/java/org/mifos/mobile/ui/activities/SettingsActivity.kt

```diff
@@ -26,11 +26,16 @@ class SettingsActivity : BaseActivity() {
     }
 
     override fun onBackPressed() {
-        super.onBackPressed()
-        if (hasSettingsChanged) {
-            ActivityCompat.finishAffinity(this)
-            val i = Intent(this, HomeActivity::class.java)
-            startActivity(i)
+        val count = supportFragmentManager.backStackEntryCount
+        if (count == 0) {
```

Comment:
What do you think about decreasing the number of lines by doing the following?

- `if (supportFragmentManager.backStackEntryCount == 0) {...}`
- `startActivity(Intent(this, HomeActivity::class.java))`

Reply:
Yes I agree, no use in creating new variables. Changes are done.